back/src/domains/establishment/adapters/PgDiscussionRepository.integration.test.ts

```diff
@@ -85,705 +91,2332 @@ describe("PgDiscussionRepository", () => {
+          rejectedDiscussion,
+          pendingDiscussion,
+          acceptedDiscussion,
+        ]);
+
+        expectToEqual(
+          await pgDiscussionRepository.__test_getDiscussionsStats(),
+          unitialStats,
+        );
+      });
+
+      describe("discussion deletion process", () => {
+        it("when id provided and discussion exist, the discussion does not exist anymore", async () => {
+          await insertDiscussions(pgDiscussionRepository, [
+            acceptedDiscussion,
+            pendingDiscussion,
+            rejectedDiscussion,
+          ]);
+
+          await pgDiscussionRepository.archiveDiscussions([
+            acceptedDiscussion.id,
+          ]);
+          expectToEqual(
+            await pgDiscussionRepository.__test_getAllDiscussions(),
+            [rejectedDiscussion, pendingDiscussion],
           );
         });
 
-        it("include only discussions answered by establishment", async () => {
+        it("when ids provided and discussions exist, discussions does not exist anymore", async () => {
+          await insertDiscussions(pgDiscussionRepository, [
+            acceptedDiscussion,
+            pendingDiscussion,
+            rejectedDiscussion,
+          ]);
+
+          await pgDiscussionRepository.archiveDiscussions([
+            acceptedDiscussion.id,
+            pendingDiscussion.id,
+          ]);
           expectToEqual(
-            await pgDiscussionRepository.getDiscussions({
-              filters: {
-                answeredByEstablishment: true,
-              },
-              limit: 5,
-            }),
-            [discussionAnsweredByEstablishment],
+            await pgDiscussionRepository.__test_getAllDiscussions(),
+            [rejectedDiscussion],
           );
         });
 
-        it("include only discussions not answered by establishment", async () => {
-          expectToEqual(
-            await pgDiscussionRepository.getDiscussions({
-              filters: {
-                answeredByEstablishment: false,
-              },
-              limit: 5,
-            }),
-            [discussionNotAnsweredByEstablishment],
+        it("when ids provided and discussions not exist, throw an error", async () => {
+          await expectPromiseToFailWithError(
+            pgDiscussionRepository.archiveDiscussions([acceptedDiscussion.id]),
+            errors.discussion.missingNotDeleted([acceptedDiscussion.id]),
+          );
+        });
+
+        it("when ids provided and one discussion doesn't exist, throw an error", async () => {
+          await insertDiscussions(pgDiscussionRepository, [rejectedDiscussion]);
+
+          await expectPromiseToFailWithError(
+            pgDiscussionRepository.archiveDiscussions([
+              acceptedDiscussion.id,
+              rejectedDiscussion.id,
+            ]),
+            errors.discussion.missingNotDeleted([acceptedDiscussion.id]),
```

Comment:
sortir l'id de la discussion non existante pour rendre l'intention des tests plus claire :
```suggestion
        
        it("when ids provided and discussions not exist, throw an error", async () => {
        const notFoundDiscussionId = acceptedDiscussion.id
          await expectPromiseToFailWithError(
            pgDiscussionRepository.archiveDiscussions([notFoundDiscussionId]),
            errors.discussion.missingNotDeleted([notFoundDiscussionId]),
          );
        });

        it("when ids provided and one discussion doesn't exist, throw an error", async () => {
         const notFoundDiscussionId = acceptedDiscussion.id
         
          await insertDiscussions(pgDiscussionRepository, [rejectedDiscussion]);

          await expectPromiseToFailWithError(
            pgDiscussionRepository.archiveDiscussions([
              notFoundDiscussionId,
              rejectedDiscussion.id,
            ]),
            errors.discussion.missingNotDeleted([notFoundDiscussionId]),
```